src/dataflow/sink/avro_ocf.rs

```diff
@@ -0,0 +1,68 @@
+// Copyright Materialize, Inc. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::fs::File;
+
+use log::error;
+use timely::dataflow::channels::pact::Pipeline;
+use timely::dataflow::operators::generic::Operator;
+use timely::dataflow::{Scope, Stream};
+
+use dataflow_types::{AvroOcfSinkConnector, Diff, Timestamp};
+use expr::GlobalId;
+use interchange::avro::Encoder;
+use repr::{RelationDesc, Row};
+
+pub fn avro_ocf<G>(
+    stream: &Stream<G, (Row, Timestamp, Diff)>,
+    id: GlobalId,
+    connector: AvroOcfSinkConnector,
+    relation_desc: RelationDesc,
+) where
+    G: Scope<Timestamp = Timestamp>,
+{
+    let schema = interchange::avro::encode_schema(&relation_desc).expect("");
+
+    stream.sink(Pipeline, &format!("avro-ocf-{}", id), move |input| {
+        let encoder = Encoder::new(&schema.to_string());
+        let parsed_schema =
+            avro::schema::Schema::parse(&schema).expect("parsing json schema failed");
+        let sink_file = File::create(&connector.path).expect("file creation failed");
```

Comment:
File creation might legitimately fail, so don't want to panic here.

Reply:
right I think more generally I need to get rid of all of the causes of panics in this code